### PR TITLE
Add module tests for storage

### LINKED ISSUE #1402: Module tests: RocksDB-related logic
Add unit and integration tests for RocksDB interactions and logic around read/write patterns, iteration, compaction behavior, and error handling. Tests should prefer mocking or ephemeral on-disk DB directories to avoid brittle external dependencies. Focus areas:
- Read/write correctness and boundary conditions
- Iteration and compaction-related logic
- Error and recovery paths when DB operations fail

This depends on the shared testing infra for RocksDB mocks/fixtures.

Acceptance criteria:
- Tests covering common and boundary RocksDB behaviors
- Tests included in the same PR as any code changes

### PR DESCRIPTION
**Pull request by henry0715-dev**
Close #1402

### CODE REVIEW

**Review by dayeon5470 — APPROVED** [review]

File: src/storage.rs [hunk 1]
```diff
@@ -1361,10 +1438,139 @@ mod tests {
         assert_eq!(results.len(), 2);
         assert_eq!(results[0].0, ts_early);
         assert_eq!(results[0].1, b"alpha");
-        assert_eq!(results[1].0, ts_late);
+        assert_eq!(results[1].0, ts_early + Duration::seconds(2));
```
Comment (dayeon5470) [inline src/storage.rs]: Since it’s already declared, it doesn’t seem necessary to change it.

**Comment by henry0715-dev** [comment]
@dayeon5470
All requested changes have been applied.

**Review by dayeon5470 — DISMISSED** [review]

**Comment by henry0715-dev** [comment]
I’ve updated the requested changes.

**Comment by dayeon5470** [comment]
`test_retain_periodically_keeps_boundary_timestamp` can be flaky because the test computes the “boundary” timestamp as now_test - retention_period, but the retention task computes its cutoff later as now_worker - retention_period. Since now_worker is typically later than now_test, the effective cutoff shifts forward, and the record written at the test’s boundary may be considered expired and deleted.

**Comment by henry0715-dev** [comment]
> `test_retain_periodically_keeps_boundary_timestamp` can be flaky because the test computes the “boundary” timestamp as now_test - retention_period, but the retention task computes its cutoff later as now_worker - retention_period. Since now_worker is typically later than now_test, the effective cutoff shifts forward, and the record written at the test’s boundary may be considered expired and deleted.

I’ve updated it.

**Comment by henry0715-dev** [comment]
> To avoid flakiness caused by the worker computing the cutoff later, using a timestamp safely inside the retention window (e.g., now - 1s) is reasonable for test stability. However, this changes what the test verifies: it no longer checks boundary preservation (== cutoff), but instead checks that non-expired data within the retention window is not deleted. It would be better to rename the test to match this intent.

The test function name has been changed from `test_retain_periodically_keeps_boundary_timestamp` to
`test_retain_periodically_keeps_within_retention_window`.

File: src/storage.rs [hunk 2]
```diff
@@ -1557,4 +1974,301 @@ mod tests {
...
+
+        let db_opts = DbOptions::default();
+        let db = Database::open(dir.path(), &db_opts).unwrap();
+        let conn_store = db.conn_store().unwrap();
+        let key = b"test_key";
+        let value = b"test_value";
+        conn_store.append(key, value).unwrap();
+        drop(db);
+
+        super::repair_db(dir.path(), 8000, 512, 8, 2, true);
```
[the remark this answers is not in the document]
Comment (henry0715-dev) [reply src/storage.rs]: I’ve applied the requested changes.

[the remark this answers is not in the document]
Comment (dayeon5470) [reply src/storage.rs]: `repair_db` takes data_dir and appends "db", so passing db_path would target <data_dir>/db/db. Using dir.path() here looks correct.

**Review by dayeon5470 — APPROVED** [review]

**Review by sophie-cluml — CHANGES_REQUESTED** [review]
Current PR description seems outdated, could you adjust your PR description please?
